soc/arc/snps_emsdp/linker.ld

```diff
@@ -39,4 +39,16 @@
 #define DCCM_SIZE		DT_REG_SIZE(DT_INST(0, arc_dccm))
 #endif
 
+#if DT_NODE_HAS_PROP(DT_INST(0, arc_xccm), reg) && \
```

Comment:
these linker snippets are the same for every soc - can this be factored out and put in a common file (maybe there's more to refactor, besides the XY memory).

Reply:
Move them to zephyr/include/arch/arc/v2/xy_mem.ld